meta_test.go

```diff
@@ -56,48 +56,173 @@ func (s *MetaSuite) TestReadMetaVersion2(c *gc.C) {
+		"term_123-23aAf/1",
+		"cs:foobar/27",
+	}
+
+	invalid := []string{
+		"/",
+		"/1",
+		"//",
+		"//2",
+		"27",
+		"owner/foo/foobar",
+		"@les/term/1",
+		"own_er/foobar",
+	}
+
+	for i, s := range valid {
+		c.Logf("valid test %d: %s", i, s)
+		meta := charm.Meta{Terms: []string{s}}
+		err := meta.Check()
+		c.Check(err, jc.ErrorIsNil)
+	}
+
+	for i, s := range invalid {
+		c.Logf("invalid test %d: %s", i, s)
+		meta := charm.Meta{Terms: []string{s}}
+		err := meta.Check()
+		c.Check(err, gc.NotNil)
+	}
+}
+
 func (s *MetaSuite) TestCheckTerms(c *gc.C) {
 	tests := []struct {
 		about       string
 		terms       []string
 		expectError string
 	}{{
 		about: "valid terms",
-		terms: []string{"term/1", "term/2", "term-without-revision"},
+		terms: []string{"term/1", "term/2", "term-without-revision", "tt/2"},
 	}, {
 		about:       "revision not a number",
 		terms:       []string{"term/1", "term/a"},
-		expectError: "invalid term name \"term/a\": must match.*",
+		expectError: `wrong term name format "a"`,
 	}, {
 		about:       "negative revision",
 		terms:       []string{"term/-1"},
-		expectError: "invalid term name \"term/-1\": must match.*",
+		expectError: "negative term revision",
 	}, {
 		about:       "wrong format",
-		terms:       []string{"term/1", "term/a/1"},
-		expectError: "invalid term name \"term/a/1\": must match.*",
+		terms:       []string{"term/1", "foobar/term/abc/1"},
+		expectError: `unknown term format "foobar/term/abc/1"`,
+	}, {
+		about: "term with owner",
+		terms: []string{"term/1", "term/abc/1"},
+	}, {
+		about: "term with owner no rev",
+		terms: []string{"term/1", "term/abc"},
 	}, {
 		about:       "term may not contain spaces",
 		terms:       []string{"term/1", "term about a term"},
-		expectError: "invalid term name \"term about a term\": must match.*",
+		expectError: `wrong term name format "term about a term"`,
 	}, {
-		about:       "term name must start with lowercase letter",
-		terms:       []string{"Term/1"},
-		expectError: `invalid term name "Term/1": must match.*`,
+		about: "term name must start with lowercase letter",
+		terms: []string{"Term/1"},
 	}, {
-		about:       "term name match the regexp",
-		terms:       []string{"term_123-23aAf/1"},
-		expectError: "invalid term name \"term_123-23aAf/1\": must match.*",
+		about: "term name match the regexp",
+		terms: []string{"term_123-23aAf/1"},
 	},
 	}
 	for i, test := range tests {
 		c.Logf("running test %v: %v", i, test.about)
 		meta := charm.Meta{Terms: test.terms}
 		err := meta.Check()
 		if test.expectError == "" {
-			c.Assert(err, jc.ErrorIsNil)
+			c.Check(err, jc.ErrorIsNil)
+		} else {
+			c.Check(err, gc.ErrorMatches, test.expectError)
+		}
+	}
+}
+
+func (s *MetaSuite) TestParseTerms(c *gc.C) {
+	tests := []struct {
+		about       string
+		term        string
+		expectError string
+		expectTerm  charm.Term
+	}{{
+		about:      "valid term",
+		term:       "term/1",
+		expectTerm: charm.Term{"", "", "term", 1},
```

Comment:
Again, I think these literals read more easily when they're tagged
with field names, so we don't have to remember the field order.